searches/binary_search.py

```diff
@@ -298,8 +298,11 @@ def binary_search_by_recursion(
 
 
 if __name__ == "__main__":
-    user_input = input("Enter numbers separated by comma:\n").strip()
-    collection = sorted(int(item) for item in user_input.split(","))
+    user_input = input("Enter numbers separated by either comma or space \n").strip()
+    if "," in user_input:
```

Comment:
This would break if there were both commas and spaces in the input, i.e. `1,000 1,001 1,002`

Reply:
Just believing that the user won't provide input in a certain way generally isn't a good way to deal with user input. If it's something that a user might reasonably do then we should still handle it somehow. At the very least, you could just prohibit the user from using both commas and spaces at the same time.